testsuite/forge/src/backend/k8s/swarm.rs

```diff
@@ -203,8 +156,12 @@ impl Swarm for K8sSwarm {
             .map(|v| v as &mut dyn Validator)
     }
 
-    fn upgrade_validator(&mut self, _id: PeerId, _version: &Version) -> Result<()> {
-        todo!()
+    fn upgrade_validator(&mut self, id: PeerId, version: &Version) -> Result<()> {
+        let validator = self
+            .validators
+            .get_mut(&id)
+            .ok_or_else(|| anyhow!("Invalid id: {}", id))?;
+        set_validator_image_tag(&validator.name(), self.versions.get(version).unwrap(), &self.helm_repo)
```

Comment:
don't unwrap here, instead return an error, we want to avoid panicking

Reply:
resolved